app/src/main/java/com/example/giftly/HomeScreen.java

```diff
@@ -170,10 +167,11 @@ public void run() {
                                                 for (int i = 0; i < events.size(); i++) {
                                                     Button button = new Button(eventList.getContext());
                                                     button.setId(i);
+
                                                     // Set the event name to lowercase
                                                     String eventName = events.get(i).getEventName().toLowerCase();
-                                                    // Capitalize the first letter of the event name
-                                                    eventName = eventName.substring(0, 1).toUpperCase() + eventName.substring(1);
+                                                    // Capitalize the first letter of the event name if it has a length
+                                                    if (eventName.length() > 1) eventName = eventName.substring(0, 1).toUpperCase() + eventName.substring(1);
```

Comment:
To make the this line code more organized, avoid doing the statement in one line,

Reply:
I believe this was overwritten by a follow-up commit